validation/src/CueVetDiagnosticsProvider.ts

```diff
@@ -0,0 +1,130 @@
+import * as vscode from 'vscode';
+import { DiagnosticProvider } from './DiagnosticsProvider';
+import { spawn } from 'child_process';
+import { writeFileSync } from 'fs';
+import { mkdtempSync, rmdirSync } from 'node:fs';
+import { join } from 'node:path';
+import { tmpdir } from 'node:os';
+import { randomBytes } from 'crypto';
+
+export class CueVetDiagnosticsProvider implements DiagnosticProvider {
+    private collection: vscode.DiagnosticCollection
+
+    private mockContext = `#Context: {
+        appRevision:    string
+        appRevisionNum: int
+        appName:        string
+        name:           string
+        namespace:      string
+        output:         _
+    }
+    
+    context: #Context & {
+        appRevision:    "v1"
+        appRevisionNum: 1
+        appName:        "dummy-app-name"
+        name:           "dummy-name"
+        namespace:      "dummy-namespace"
+        output:         {}
+    }`;
+
+    private tempDirectory: string | undefined;
+
+    constructor(collection: vscode.DiagnosticCollection) {
+        this.collection = collection;
+    }
+
+    activate(): void {
+        const directory = mkdtempSync(join(tmpdir(), 'vela-vscode-extension-'));
+        console.log(directory);
+        this.tempDirectory = directory;
+    }
+
+    deactivate(): void {
+        if (this.tempDirectory) {
+            rmdirSync(this.tempDirectory);
```

Comment:
Bug that needs fixing before this is merged: these directories aren't cleaned up.